### PR TITLE
Allow to dump public keys in CLI

### LINKED ISSUE #1810: Print wallet public key command in neo-go cli
It would be a convinient feature if I might get a wallet's public key with neo-go CLI.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1810 .
```
> bin/neo-go wallet dump -w .docker/wallets/wallet1_solo.json --address NVNvVRW5Q5naSx2k2iZm7xRgtRNGuZppAK
1 out of 1 multisig contract:
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2

> bin/neo-go wallet dump -w .docker/wallets/wallet1_solo.json --address NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY
3 out of 4 multisig contract:
02103a7f7dd016558597f7960d27c516a4394fd968b9e65155eb4b013e4040406e
02a7bc55fe8684e0119768d104ba30795bdcc86619e864add26156723ed185cd62
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2
03d90c07df63e690ce77912e10ab51acc944b66860237b608c4f8f8309e71ee699

> bin/neo-go wallet dump -w .docker/wallets/wallet1_solo.json --address NbTiM6h8r99kpRtb428XcsUk1TzKed2gTc
Simple signature contract:
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2
```

### CODE REVIEW

**Comment by fyrchik** [comment]
Depending on the use-case we may want to print our public key for multisig contracts.

File: cli/wallet/wallet.go [hunk 1]
```diff
@@ -118,6 +119,10 @@ func NewCommands() []cli.Command {
 				Flags: []cli.Flag{
 					walletPathFlag,
 					decryptFlag,
+					cli.StringFlag{
+						Name:  "address",
```
Comment (roman-khimov) [inline cli/wallet/wallet.go:123]: I'd interpret this parameter as "dump data for this address only", which actually is a useful thing to have, but it's a different functionality.

This option should be something as `extract-keys, k` or (if we don't use JSON output format that is expected from the `dump`) a separate command.

Comment (fyrchik) [reply cli/wallet/wallet.go:123]: I think `dump-keys [-a ..]` is ok.

**Comment by fyrchik** [comment]
All addresses:
```
> bin/neo-go wallet dump-keys -w .docker/wallets/wallet1_solo.json
NbTiM6h8r99kpRtb428XcsUk1TzKed2gTc (simple signature contract):
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2

NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY (3 out of 4 multisig contract):
02103a7f7dd016558597f7960d27c516a4394fd968b9e65155eb4b013e4040406e
02a7bc55fe8684e0119768d104ba30795bdcc86619e864add26156723ed185cd62
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2
03d90c07df63e690ce77912e10ab51acc944b66860237b608c4f8f8309e71ee699

NVNvVRW5Q5naSx2k2iZm7xRgtRNGuZppAK (1 out of 1 multisig contract):
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2
```

Single address:
```
> bin/neo-go wallet dump-keys -w .docker/wallets/wallet1_solo.json --address NbTiM6h8r99kpRtb428XcsUk1TzKed2gTc
NbTiM6h8r99kpRtb428XcsUk1TzKed2gTc (simple signature contract):
02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2
```

File: cli/wallet/wallet.go (hunk 1 above)
Comment (roman-khimov) [reply cli/wallet/wallet.go:131]: Yep, except `-a` is not supported at the moment.

**Review by roman-khimov — COMMENTED** [review]
Looks good, but short form should also be supported.

File: cli/wallet/wallet.go [hunk 2]
```diff
@@ -120,6 +121,18 @@ func NewCommands() []cli.Command {
 					decryptFlag,
 				},
 			},
+			{
+				Name:   "dump-keys",
+				Usage:  "dump public keys for account",
```
Comment (AnnaShaleva) [inline cli/wallet/wallet.go:126]: Add usage text?

**Comment by AnnaShaleva** [comment]
Don't we want to support printing unencrypted WIFs? It may be useful for cases when we change verification scripts (like in #1824).

**Comment by roman-khimov** [comment]
> Don't we want to support printing unencrypted WIFs? It may be useful for cases when we change verification scripts (like in #1824).

`wallet dump -d`?

**Comment by fyrchik** [comment]
For regeneration we already have test in `pkg/wallet/regenerate_test.go`. But printing WIFs can nonetheless be useful.

**Comment by AnnaShaleva** [comment]
> > Don't we want to support printing unencrypted WIFs? It may be useful for cases when we change verification scripts (like in #1824).
> 
> `wallet dump -d`?

IIUC, it does not print unencrypted WIF (encrypted only).

> For regeneration we already have test in `pkg/wallet/regenerate_test.go`.

Somehow I missed that...

File: cli/wallet/wallet.go [hunk 3]
```diff
@@ -120,6 +121,18 @@ func NewCommands() []cli.Command {
...
+				Action: dumpKeys,
+				Flags: []cli.Flag{
+					walletPathFlag,
+					cli.StringFlag{
+						Name:  "address",
+						Usage: "adress to print public keys for",
```
Comment (roman-khimov) [inline cli/wallet/wallet.go:132]: address

**Comment by fyrchik** [comment]
Actually, we already have `wallet export`:
```
> bin/neo-go wallet export -w .docker/wallets/wallet1_solo.json -d NbTiM6h8r99kpRtb428XcsUk1TzKed2gTc
Enter password >
KxyjQ8eUa4FHt3Gvioyt1Wz29cTUrE4eTqX3yFSk1YFCsPL8uNsY
```

I think it is better to merge them in one command, may be add `export --public-only`.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**